### DISCUSSION TITLE
Migrating Windows Forms application to UNO

### DISCUSSION THREAD

**Discussion by jonatanarr** [unoplatform/uno#9376] [Q&A]
Hello,

We have a Windows Forms application with a lot of C# code and we want to migrate it to Linux. We are looking at the options and UNO Platform seems the best. Our application makes 2D drawings on the screen, so we need hardware acceleration, for this I think SKIA can be useful. However, in the future we want to display 3D objects on the screen. Will it be possible to do this in UNO Platform with some library? Has anyone done something similar?

Thank you!

**Comment by francoistanguay** [comment]
Pretty sure you should be able to use Angle? Also seen various OPenGLT on GTK# attempts in the community.

Will try to check with the team.

**Reply by jeromelaban** [reply]
The underlying support for OpenGL with Skia uses [Silk.NET](https://dotnet.github.io/Silk.NET/). As long as the underlying platform supports OpenGL (and that OpenGL Skia support is enabled), it should be possible to use the `Silk.NET.OpenGL` namespace to render 3D content.

We do not yet have the hooks to expose a reliable "render loop" for applications to render content, though. Uno will likely need to provide an OpenGL rendering surface that hooks onto the proper rendering stage, and uses the appropriate viewport and compositing so it looks right in the app.

Let's track this in https://github.com/unoplatform/uno/issues/9405
